### DISCUSSION TITLE
Include parts of a template/additional snippet based on day of the week

### DISCUSSION THREAD

**Discussion by Gewerd-Strauss** [SilentVoid13/Templater#843] [Help] [answered]
Hello,

For my daily-note, I currently fill in a lot of additional information for recurring events that are weekday-specific. So far, this was manageable, but it becomes a bit too much to upkeep with additional normal maintenance going on. 
Nonetheless, it is required for me, and thus I'd like to know if I can have templater include sections based on the filename via `<% this.app.workspace.getActiveFile().basename %>`

The included sections range from a single line to several lines long, containing headers and further formatting.

How would this be possible? 

Thank you.
Sincerely,
~Gw

---
Functionally this is identical to #828

**Comment by AB1908** [comment]
You'll want to write scripts starting with `<%*` to execute general purpose JavaScript. Look at other showcase discussions for examples. For managing your templates, use `tp.file.include` and break them into parts so that you can include whatever is necessary.

**Reply by Gewerd-Strauss** [reply]
This is my current attempt. I am not sure why/how I would employ JS here with `<%*` 
    
    <% await tp.file.include("[[Monday - Food]]") %>
    <% await tp.file.include(""[[<% await tp.file.include( tp.date.now("dddd") " - Food") %>]]"") %>

The upper line alone can execute, the lower is what I am trying o get to essentially. It might be wrong, but I am not sure why/how to fix it.

**Reply by Gewerd-Strauss** [reply]
@AB1908 I am sorry for pinging you, I cannot figure it out.

I have had a couple of days attempting to understand, and I have to admit I don't. I simply don't. I don't get why I would have to use JS instead of the more "standard" `<%%>`-syntax, I fail to understand how to do so.
After four days of miserably failing; I tried attempting it via my initial idea, and a lot of tinkering later I came up with this:

`[[Day - 0<% tp.date.now("d dddd")%>]]` - this can print the contents of the following line just well:
`[[Day - 04 Thursday]]`

and

`<% tp.file.include("[[Day - 04 Thursday]]") %>` - this can include the contents of the file just well

I just cannot figure out how to combine them properly so that this actually works :/ I am aware that this might be completely wrong, but going JS was somehow even more unsuccessful for me.

**Reply by Gewerd-Strauss** [reply]
Jesus Christ sometimes I miss simple shit. I ended up settling on

    <%  tp.file.include("{{Day - 0" + tp.date.now("d dddd",0,this.app.workspace.getActiveFile().basename,"DD.MM.YYYY") + "}}") %>

Only that I spent another 90 minutes failing to figure out the damn "+" for concatenation :P

**Reply by AB1908** [reply]
Hi Gewerd, sorry I haven't replied sooner. You can use a couple of different approaches but I'll talk about only one here. You can have 7 different "partial templates" based on day of the week like so: `monday.md`, `tuesday.md`, etc. and in your daily note template, you can check for weekday and insert the correct template.

```js
<%*
const weekday = moment(<date goes here>).weekday();
switch (weekday) {
  case 0: await tp.file.include("[[monday]]"); break;
// rest of similar logic here
}
%>
```

**Reply by Gewerd-Strauss** [reply]
No problem, I have been busy myself. In fact, I have been too busy, cuz after figuring out the solution (cf. https://github.com/SilentVoid13/Templater/discussions/828#discussioncomment-3830517), I completely forgot to update these discussions :/
I ended up taking a different solution path to what you outlined, and with a bit of brain pain and help from reddit I made it work (didn't know how to concatenate strings properly :P).

**Comment by Gewerd-Strauss** [comment, marked as answer]
The solution: https://github.com/SilentVoid13/Templater/discussions/828#discussioncomment-383051
